diff/diff.go

```diff
@@ -58,10 +70,158 @@ func Get(config *Config, s1, s2 *openapi3.T) (*Diff, error) {
+The x-since-date extension should be set on path or operations level. Extension set on the operations level overrides teh value set on path level.
+If such path doesn't have the x-since-date extension, its value is default "2000-01-01"
+If there are same paths with the same x-since-date value, then function returns error.
+The format of the x-since-date is the RFC3339 full-date format
+
+Note that Get expects OpenAPI References (https://swagger.io/docs/specification/using-ref/) to be resolved.
+References are normally resolved automatically when you load the spec.
+In other cases you can resolve refs using https://pkg.go.dev/github.com/getkin/kin-openapi/openapi3#Loader.ResolveRefsIn.
+*/
+func GetPathsDiff(config *Config, s1, s2 []load.OpenAPISpecInfo) (*Diff, *OperationsSourcesMap, error) {
+	state := newState()
+	result := newDiff()
+	var err error
+	paths1, operationsSources1, err := mergedPaths(s1)
+	if err != nil {
+		return nil, nil, err
+	}
+	paths2, operationsSources2, err := mergedPaths(s2)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if result.PathsDiff, err = getPathsDiff(config, state, *paths1, *paths2); err != nil {
+		return nil, nil, err
+	}
+
+	if result.EndpointsDiff, err = getEndpointsDiff(config, state, *paths1, *paths2); err != nil {
+		return nil, nil, err
+	}
+
+	operationsSources := *operationsSources1
+	for k, v := range *operationsSources2 {
+		operationsSources[k] = v
+	}
+	return result, &operationsSources, nil
+}
+
+func mergedPaths(s1 []load.OpenAPISpecInfo) (*openapi3.Paths, *OperationsSourcesMap, error) {
+	result := make(openapi3.Paths, 0)
+	operationsSources := make(OperationsSourcesMap)
+	for _, s := range s1 {
+		for path, pathItem := range s.Spec.Paths {
+			p := result.Find(path)
+			if p == nil {
+				result[path] = pathItem
+				for _, opItem := range pathItem.Operations() {
+					operationsSources[opItem] = s.Url
+				}
+				continue
+			}
+
+			for op, opItem := range pathItem.Operations() {
+				oldOperation := p.GetOperation(op)
+				if oldOperation == nil {
+					p.SetOperation(op, opItem)
+					operationsSources[opItem] = s.Url
+					continue
+				}
+
+				oldSince, err := sinceDateFrom(*p, *oldOperation)
+				if err != nil {
+					return nil, nil, fmt.Errorf("Invalid %s extension value in %s(%s %s), %w", SinceDateExtension, operationsSources[oldOperation], op, path, err)
+				}
+				newSince, err := sinceDateFrom(*pathItem, *opItem)
+				if err != nil {
+					return nil, nil, fmt.Errorf("Invalid %s extension value in %s(%s %s), %w", SinceDateExtension, s.Url, op, path, err)
+				}
+				if newSince.After(oldSince) {
+					p.SetOperation(op, opItem)
+					operationsSources[opItem] = s.Url
+				}
+
+				if newSince == oldSince {
+					return nil, nil, fmt.Errorf("Multiple endpoints found in %s(%s %s) and %s(%s %s). Add the %s extension with ordered values to operations to specify its order.", operationsSources[oldOperation], op, path, s.Url, op, path, SinceDateExtension)
```

Comment:
Please fix this: error strings should not end with punctuation or newlines (ST1005)go-staticcheck

Reply:
Fixed this and other staticheck issues